lib/concurrent/edge/lock_free_linked_set/node.rb

```diff
@@ -0,0 +1,72 @@
+require 'concurrent/edge/atomic_markable_reference'
+
+module Concurrent
+  module Edge
+    class LockFreeLinkedSet
+      class Node < Synchronization::Object
+        include Comparable
+
+        attr_reader :Data, :SuccessorReference, :Key
```

Comment:
Yeah I can do that. TBH I don't know how I like the uppercase convention for final fields. It makes the code reading awkward IMO. I wonder if there is a more explicit way of doing it that doesn't require semantic casing. What are you thoughts?